### PR TITLE
extended get and set handling of color and brightness

### PR DESCRIPTION
**Pull request by mimez**
Hey, thanks for your lib. i used it but it was a limited for my purposes. I wanted to control my philips hue lights over http but the hue lights need specific requests. i extended your lib, please check it.

### CODE REVIEW

**Comment by QuickSander** [comment]
Hi Michael,

Thanks for your extensions. Looks good!

As I’m currently on a business trip to Taiwan, is it Oké if I merge the request once I’m back home?

Kind regards,
Sander van Woensel

**Comment by mimez** [comment]
Hey Sander,

no problem! take your time and have a good trip.

cheers

File: README.md [hunk 1]
```diff
@@ -212,6 +274,18 @@ remove the brightness component from the config.
...
+| Key | Description |
+| --- | --- |
+| `url` | URL to retrieve brightness status |
+| `bodyRegEx` _(optional)_ | Regular expression to extract the brightness out of a response body. Example: "bri":([0-9]+) |
+
+#### Color url object
+| Key | Description |
+| --- | --- |
+| `url` | URL to set color status |
+| `body` | relevant, if the http_method is put/post. this body will be sent to the url. you can use following placeholders: %s (hex-rgb), %xy-x, $xy-y. Example: {"xy": [%xy-x,%xy-y]}  |
```
Comment (QuickSander) [inline README.md]:
```suggestion
| `body` | relevant, if the http_method is put/post. this body will be sent to the url. you can use following placeholders: %s (hex-rgb), %xy-x, %xy-y. Example: {"xy": [%xy-x,%xy-y]}  |
```

File: index.js [hunk 2]
```diff
@@ -117,10 +118,22 @@ function HttpPushRgb(log, config) {
...
+        }
+        if (typeof config.brightness.url === 'object') {
+            this.brightness.set_url.url = config.brightness.url.url || this.brightness.status.url;
+            this.brightness.set_url.body = config.brightness.url.body || '';
+        } else {
+            this.brightness.set_url.url = config.brightness.url || this.brightness.status.url;
+            this.brightness.set_url.body = '';
+        }
         this.brightness.http_method    = config.brightness.http_method    || this.http_method;
+        this.brightness.max = config.brightness.max || 100;
```
Comment (QuickSander) [inline index.js]: Can you make this '100' into a constant defined at the top?

File: README.md [hunk 3]
```diff
@@ -200,9 +248,23 @@ remove the brightness component from the config.
...
+| `status` | URL to get RGB current brightness or a [brightness status object](#brightness-status-object). |
+| `url` | URL to set the current brightness of the RGB device or a [brightness url object](#brightness-url-object). |
 | `http_method` _(optional)_ | The brightness specific HTTP method for set requests. If omitted defaults to `http_method` as specified in the root structure |
+| `max`  _(optional)_ | This value specifies the maximum Integer for brightness. For example, Philips Hue returns 254 when brightness is at 100 % (default: 100)|
+
+#### Brightness status object
+| Key | Description |
+| --- | --- |
+| `url` | URL to retrieve brightness status |
+| `bodyRegEx` _(optional)_ | Regular expression to extract the brightness out of a response body. Example: "bri":([0-9]+) |
```
Comment (QuickSander) [inline README.md]:
```suggestion
| `bodyRegEx` _(optional)_ | Regular expression to extract the brightness out of a response body. Example: `"bri":([0-9]+)` |
```

File: README.md [hunk 4]
```diff
@@ -200,9 +248,23 @@ remove the brightness component from the config.
...
+
+#### Brightness url object
+| Key | Description |
+| --- | --- |
+| `url` | URL to set brightness status |
+| `body` | relevant, if the http_method is put/post. this body is sent to the url. you can use a placeholder (%s) within the body. example: {"bri": %s}  |
```
Comment (QuickSander) [inline README.md]:
```suggestion
| `body` | relevant, if the http_method is put/post. this body is sent to the url. you can use a placeholder (%s) within the body. example: `{"bri": %s}`  |
```

**Comment by mimez** [comment]
thanks for your review. i added a constant. hopefully it is now okay for you.

**Comment by QuickSander** [comment]
I will merge as soon as I am back in The Netherlands. 

Kind regards

**Comment by mimez** [comment]
thanks dude!

**Pull request merged**